### PR TITLE
feat(task): implement `DocumentationTask`

### LINKED ISSUE #22: Flesh out DocumentTask implementation
The first task we want to support is the `DocumentTask`. The bare bones as well as basic `Task` API definition can be found here: https://github.com/bitpicky/dbt-sugar/tree/main/dbt_sugar/core/task

In it's most basic form it should:
- know what dbt model you want to document
- check the table in prod or user's env for the table structure (col names)
- generate an empty yaml file.
- write this back into the proper schema files.

Very quickly after, we will want to make sure we can ask for user input. This means in this issue we should define the API between the document task and what it will require from the CLI front end.

It might ask for the following:
- show cols that are documented
- ask user which one they might want to document
- collect the definitions
- submit that back to the documentation object/task.

### PR DESCRIPTION
**Pull request by virvirlopez**
# Description
Fleshes out `DocumentationTask` by:
- adding some common methods to all tasks in `BaseTask`
- adding documentation specific methods to `DocumentationTask`

# How was the change tested
I have added tests to cover the functionality.

# Checklist
Closes #22 
(Ideally, all boxes are checked by the time we merged the PR, if you don't know how to do any of these don't hesitate to say so in the PR and we'll help you out.)

- [x] I formatted my PR name according to [CONTRIBUTING.md](CONTRIBUTING.md)
- [ ] I added a news fragment to help populating the changelog as encouraged in [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added "Closes #<issue_number>" in the "Issue Information" section (if no issue, feel free to tick thick the box anyway).

### CODE REVIEW

File: dbt_sugar/core/task/base.py [hunk 1]
```diff
@@ -1,10 +1,162 @@
...
 
 class BaseTask(abc.ABC):
     """Sets up basic API for task-like classes."""
 
+    def __init__(self) -> None:
+        self.repository_path = Path().absolute()
+        self.dbt_definitions: Dict[str, str] = {}
+        self.save_all_descriptions()
+
+    def get_column_description_from_dbt_definitions(self, column_name: str) -> str:
```
Comment (bastienboutonnet) [inline dbt_sugar/core/task/base.py:22]: Do you think it makes sense to have all of these methods in the base task? The `BaseTask` is supposed to be an abstract class so to me the concrete implementations should live in `DocumentTask` no?

It is my understanding that you put abstractmethods in an abstract class to define the minimum viable methods (generally the ones expected by the task factory or the rest of your system) in there as non implemented ones and then you flesh out the concrete classes with whatever you need to have in there.

Have I been thinking about this all wrong?

File: tests/base_task_test.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
+import pytest
+
+from dbt_sugar.core.task.base import COLUMN_NOT_DOCUMENTED
+from dbt_sugar.core.task.doc import DocumentationTask
+
+
+def __init_descriptions():
+    base_task = DocumentationTask(None)
```
Comment (bastienboutonnet) [inline tests/base_task_test.py:8]: Not sure why you call this `base_task` since you're testing the documentation task I guess you could use this as the variable name no?

**Review by bastienboutonnet — COMMENTED** [review]
I have some questions around the placement of the code.

Also, in general, I find the docstrings a but too vague about what they do. I think either eitehr a case of finding more concrete naming or using a few more examples in the docstring.

For example documenting `content` with "content of the yaml` doen't really help much and I would have to look at the code. 

We don't have to have perfect docstrings now but something maybe for the future to think about. I would say unless you feel like you want to rework them now, we should make a ticket to document the methods more nicely later.

File: dbt_sugar/core/task/base.py (hunk 1 above)
Comment (virvirlopez) [reply dbt_sugar/core/task/base.py:22]: these methods are going to be used all the time in different tasks because they are the basics commands, that is the reason that was in a separate script if we add them into DocumentTask is going to be a lot of replication.

File: tests/base_task_test.py (hunk 2 above)
Comment (bastienboutonnet) [reply tests/base_task_test.py:8]: On this one I would still rename `base_task` to `documentation_task` or `doc_task` to avoid people (or us wondering why).

File: dbt_sugar/core/task/base.py (hunk 1 above)
Comment (bastienboutonnet) [reply dbt_sugar/core/task/base.py:22]: Yeah I get that. If we are for sure going to need them in **all** of the tasks then I guess it can stay there. We might want to break down the base classes more into progressively more focused stuff when we come up with more tasks down the line.

File: dbt_sugar/core/clients/yaml_helpers.py [hunk 3]
```diff
@@ -24,4 +24,15 @@ def open_yaml(path: Path) -> Dict[str, Any]:
             if yaml_dict:
                 return yaml_dict
             raise YAMLFileEmptyError(f"The following file {path.resolve()} seems empty.")
-    raise FileNotFoundError(f"File {path.resolve()} was not found.")
```
Comment (virvirlopez) [inline dbt_sugar/core/clients/yaml_helpers.py:27]: I have to remove this because when we were creating a new file was failing.

File: dbt_sugar/core/task/doc.py [hunk 4]
```diff
@@ -11,8 +19,125 @@ class DocumentationTask(BaseTask):
...
+            target_name="dev",
+        )
+        dbt_profile.read_profile()
+        dbt_credentials = dbt_profile.profile
+        if not dbt_credentials:
+            logger.info("Not able to locate DBT profile.")
+            exit(1)
+        return dbt_credentials
 
     def run(self) -> int:
```
Comment (virvirlopez) [inline dbt_sugar/core/task/doc.py:38]: bastien, I dont understand why we have to return int in the run command.

Comment (bastienboutonnet) [reply dbt_sugar/core/task/doc.py]: it's something it tend to do for runners. I usually return 0. Allows to do things like returning non 0 and crash apps. This video got me into the habit of doing that: https://www.youtube.com/watch?v=Kur4Y7j4vac

**Review by bastienboutonnet — COMMENTED** [review]
Ok I think this is in a really close to mergeable state. I have a few questions if you can answer them and make changes if needed that would be awesome. If any of my requests or changes don't make sense do let me know, I also need to learn.

File: dbt_sugar/core/task/doc.py [hunk 5]
```diff
@@ -11,8 +19,120 @@ class DocumentationTask(BaseTask):
     """
 
     def __init__(self, flags: FlagParser) -> None:
-        ...
+        super().__init__()
+        self.flags = flags
+
+    def load_dbt_credentials(self) -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/task/doc.py:25]: we should not put any functionality in the __init__

File: dbt_sugar/core/main.py [hunk 6]
```diff
@@ -66,17 +67,17 @@ def check_and_print_version() -> str:
...
     "-m", "--model", help="dbt model name to document", type=str, default=None
 )
+document_sub_parser.add_argument(
+    "-s", "--schema", help="database schema where the model is.", type=str, default=None
+)
 
 
 # task handler
-def handle(parser: argparse.ArgumentParser, test_cli_args: List[str] = list()) -> int:
+def handle(parser: argparse.ArgumentParser, test_cli_args: List[str] = list()) -> None:
```
[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/main.py]: I will put the int return, I didn't understand it so I removed it

File: dbt_sugar/core/task/doc.py [hunk 7]
```diff
@@ -11,8 +19,120 @@ class DocumentationTask(BaseTask):
...
+        """Method to load the DBT profile credentials."""
+        dbt_profile = DbtProfile(
+            project_name="default",
```
[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/task/doc.py]: will do!

File: dbt_sugar/core/main.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/main.py]: I just check out that you have fixed it. Thanks!

**Review by bastienboutonnet — DISMISSED** [review]
I think this is good to go. Just saying this again but we're introducing a 7% test coverage decrease because of some corner cases that I think we don't test for. I think it's ok for now though we can revisit test coverage later.

File: dbt_sugar/core/task/doc.py [hunk 8]
```diff
@@ -11,8 +19,120 @@ class DocumentationTask(BaseTask):
...
+            target_name="dev",
```
[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/task/doc.py]: Bastien, I been checking and i don't see what more we can test that makes sense in the doc task, I have added a couple of tests.

[the remark this answers is not in the document]
Comment (virvirlopez) [reply dbt_sugar/core/task/doc.py]: https://codecov.io/gh/bitpicky/dbt-sugar/pull/36/src/dbt_sugar/core/task/doc.py

**Review by bastienboutonnet — APPROVED** [review]
LGTM 🥳
